src/rust/engine/process_execution/src/remote.rs

```diff
@@ -195,29 +228,39 @@ impl super::CommandRunner for CommandRunner {
                               "Future-Delay errored at operation result polling for {}, {}: {}",
                               operation_name, description, e
                             )
-                          })
-                          .and_then(move |_| {
+                          }).and_then(move |_| {
                             future::done(
                               operations_client
-                                .get_operation_opt(&operation_request, command_runner3.call_option())
-                                .or_else(move |err| {
+                                .get_operation_opt(
+                                  &operation_request,
+                                  command_runner3.call_option(),
+                                ).or_else(move |err| {
                                   rpcerror_recover_cancelled(operation_request.take_name(), err)
-                                })
-                                .map(OperationOrStatus::Operation)
+                                }).map(OperationOrStatus::Operation)
                                 .map_err(rpcerror_to_string),
                             ).map(move |operation| {
-                              future::Loop::Continue((operation, iter_num + 1))
-                            })
-                              .to_boxed()
-                          })
-                          .to_boxed()
+                              future::Loop::Continue((history, operation, iter_num + 1))
+                            }).to_boxed()
+                          }).to_boxed()
                       }
                     }
                   }
                 })
-            })
-          })
-          .to_boxed()
+              },
+            )
+          }).map(move |resp| {
+            let mut attempts = String::new();
+            for (i, attempt) in resp.execution_attempts.iter().enumerate() {
+              attempts += &format!("\nAttempt {}: {:?}", i, attempt);
+            }
+            debug!(
+              "Finished remote exceution of {} after {} attempts: Stats: {}",
```

Comment:
If it were me, I think I'd add a newline after `attempts:`, and another one after `Stats: `. But totally a style choice.

Reply:
There will already be a newline after stats because we construct each attempt with a leading `\n`; I don't buy one after attempts, as it's just an int :)